code-challenges.test.js

```diff
@@ -25,27 +27,154 @@ const colors2 = [
+}
+
+// There is some discussion that the above implementation does not provide fair distributions. The Fisher-Yates Algorithm and variations are said to provide more consistently random results.
+
+// Create a function isherYates that removes the first element of an array and then shuffles the remaining elements
+// Input: an array
+// Output: an array
+// Removing the first element of the array can be accomplished using .slice(1)
+// Create a for loop to iterate through the sliced array
+      // Initialize variables i and j
+      // Begin at the end of the array and work towards the beginning
+      // i is used as the index of the main element
+      // during each pass, j is defined as a random number between the beginning and considered element
+      // swap the elements at indexes i and j
+// Return the resulting array
+
+// const isherYates = (array) => {
+//       array = array.slice(1)
+//       for(let [i, j] = [array.length - 1, 0]; i > 0; i --) {
+//             j = Math.floor(Math.random() * (i + 1))
+//             [array[j], array[i]] = [array[i], array[j]]
+//       }
+//       return array
+// }
+
+// Refactored to modify i and j in an array within the third argument passed to for()
+// Throws typeError if the input is not an array
+const isherYates = (array) => {
+      if(! Array.isArray(array)){
+            throw TypeError('please pass an array')
+      }
+      array = array.slice(1)
+      for(let [i, j] = [array.length - 1, 0]; i > 0; [i --, j = Math.floor(Math.random() * (i + 1))]) {[array[j], array[i]] = [array[i], array[j]]}
+      return array
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.527 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 1.67s.
+
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("talleyVotes", () => {
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
+it("takes in an object that contains up votes and down votes and returns the end tally", () => {
+      expect(talleyVotes(votes1)).toEqual(11)
+      expect(talleyVotes(votes2)).toEqual(-31)
+})
+})
+
+// ReferenceError: talleyVotes is not defined
 
 // b) Create the function that makes the test pass.
 
+// Create a function talleyVotes to count the votes
+// Input: an object keys for upVotes and downVotes; values associated with both keys are integers
+// Output: a signed integer indicating the total of votes; negative indicates there were more downVotes
+// Access the key:value pairs to perform simple arithmetic
+// Throws type error if any of the elements are not configured properly
+// Number.isSafeInteger provides Boolean response on whether a value is a safe integer
```

Comment:
Ooo!  Nice work on coming up with some edge cases!  It could be cool to add a test that will check that as well for an added stretch.